src/server/ProcessList.cpp

```diff
@@ -30,51 +27,29 @@ using namespace Microsoft::Console::Interactivity;
 [[nodiscard]] HRESULT ConsoleProcessList::AllocProcessData(const DWORD dwProcessId,
                                                            const DWORD dwThreadId,
                                                            const ULONG ulProcessGroupId,
-                                                           _In_opt_ ConsoleProcessHandle* const pParentProcessData,
                                                            _Outptr_opt_ ConsoleProcessHandle** const ppProcessData)
 {
-    FAIL_FAST_IF(!(ServiceLocator::LocateGlobals().getConsoleInformation().IsConsoleLocked()));
+    assert(ServiceLocator::LocateGlobals().getConsoleInformation().IsConsoleLocked());
 
     auto pProcessData = FindProcessInList(dwProcessId);
-    if (nullptr != pProcessData)
+    if (pProcessData)
     {
-        // In the GenerateConsoleCtrlEvent it's OK for this process to already have a ProcessData object. However, the other case is someone
-        // connecting to our LPC port and they should only do that once, so we fail subsequent connection attempts.
-        if (nullptr == pParentProcessData)
-        {
-            return E_FAIL;
-            // TODO: MSFT: 9574803 - This fires all the time. Did it always do that?
-            //RETURN_HR(E_FAIL);

```

Comment:
There's only two callers. `ApiDispatchers::ServerGenerateConsoleCtrlEvent` always calls this with a valid `pParentProcessData` and won't ever run into this condition. `IoDispatchers::ConsoleHandleConnectionRequest` calls this method with a `nullptr` instead. This indicates that this error triggers, because we're getting redundant connection requests for the same process ID.

The new code will now always return `E_FAIL`, because the former, `ApiDispatchers::ServerGenerateConsoleCtrlEvent`, actually only calls this function if the `dwProcessId` is not in the list:
https://github.com/microsoft/terminal/blob/437b5ac595915830194498eca8825e6037e288c0/src/server/ApiDispatchersInternal.cpp#L83-L84

Reply:
No it does get called by `IoDispatchers::ConsoleHandleConnectionRequest`. It sets `pParentProcessData` to `nullptr` which then invokes this branch. It does that so that `AllocProcessData` returns `S_OK` instead of `E_FAIL` if the given process ID already exists.

...but that practically makes the `pParentProcessData` parameter a `bool allowDuplicateProcessId` and I didn't particularly like that idea, when the caller might as well just filter the `E_FAIL` message themselves. This way the `AllocProcessData` is now way way simpler: It either sucessfully inserts a new process into the list or it fails. Simple as that! I felt like that would be a good idea.